src/core/src/op/scatter_nd_update.cpp

```diff
@@ -2,134 +2,134 @@
-                                                   arg2->get_data_ptr<ET>(),
-                                                   out->get_data_ptr<ET>(),
-                                                   arg0->get_shape(),
-                                                   arg1->get_shape(),
-                                                   arg2->get_shape());
-    } else if (arg1->get_element_type() == element::i32) {
-        ov::reference::scatterNdUpdate<T, int32_t>(arg0->get_data_ptr<ET>(),
-                                                   arg1->get_data_ptr<int32_t>(),
-                                                   arg2->get_data_ptr<ET>(),
-                                                   out->get_data_ptr<ET>(),
-                                                   arg0->get_shape(),
-                                                   arg1->get_shape(),
-                                                   arg2->get_shape());
-    } else {
-        OPENVINO_THROW("Unexpected type ",
-                       arg1->get_element_type().c_type_string(),
-                       " for ScatterNDUpdate evaluate method.");
-    }
-
-    return true;
+    return std::make_shared<ScatterNDUpdate>(new_args.at(util::ScatterNDBase::INPUTS),
+                                             new_args.at(util::ScatterNDBase::INDICES),
+                                             new_args.at(util::ScatterNDBase::UPDATES));
 }
 
-bool evaluate_scatter(const HostTensorPtr& arg0,
-                      const HostTensorPtr& arg1,
-                      const HostTensorPtr& arg2,
-                      const HostTensorPtr& out) {
-    bool rc = true;
-
-    switch (out->get_element_type()) {
-        OPENVINO_TYPE_CASE(evaluate_scatter, i32, arg0, arg1, arg2, out);
-        OPENVINO_TYPE_CASE(evaluate_scatter, i64, arg0, arg1, arg2, out);
-        OPENVINO_TYPE_CASE(evaluate_scatter, u32, arg0, arg1, arg2, out);
-        OPENVINO_TYPE_CASE(evaluate_scatter, u64, arg0, arg1, arg2, out);
-        OPENVINO_TYPE_CASE(evaluate_scatter, f16, arg0, arg1, arg2, out);
-        OPENVINO_TYPE_CASE(evaluate_scatter, f32, arg0, arg1, arg2, out);
-        OPENVINO_TYPE_CASE(evaluate_scatter, boolean, arg0, arg1, arg2, out);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
-}
-}  // namespace
-}  // namespace scatter
-
-bool op::v3::ScatterNDUpdate::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool ScatterNDUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v3_ScatterNDUpdate_evaluate);
-    OPENVINO_ASSERT(!inputs.empty());
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    OPENVINO_ASSERT(validate_host_tensor_vector(inputs, 3));
-    OPENVINO_ASSERT(validate_host_tensor_vector(outputs, 1));
-    OPENVINO_SUPPRESS_DEPRECATED_END
-
-    return scatter::evaluate_scatter(inputs[0], inputs[1], inputs[2], outputs[0]);
+    OPENVINO_ASSERT(inputs.size() == 3);
```

Comment:
As in other `Scatters` maybe `shape_infer` should be used (better validation) but maybe is not requried.

Reply:
`shape_infer` is to "heavy" in this context.